server/app/services/question/ActiveAndDraftQuestions.java

```diff
@@ -94,4 +159,39 @@ public Optional<QuestionDefinition> getDraftQuestionDefinition(String name) {
         ? versionedByName.get(name).second()
         : Optional.empty();
   }
+
+  public ReferencingPrograms getReferencingPrograms(String name) {
+    return ReferencingPrograms.builder()
+        .setActiveReferences(referencingActiveProgramsByName.getOrDefault(name, ImmutableSet.of()))
+        .setDraftReferences(
+            referencingDraftProgramsByName.map(r -> r.getOrDefault(name, ImmutableSet.of())))
+        .build();
+  }
+
+  /** Contains sets of programs in the active and draft versions that reference a given question. */
+  @AutoValue
+  public abstract static class ReferencingPrograms {
+
+    /**
+     * Returns a set of references to the question in the DRAFT version. This returns
+     * Optional.empty() if there are no edited programs or questions in the DRAFT version.
+     */
+    public abstract Optional<ImmutableSet<ProgramDefinition>> draftReferences();
```

Comment:
suggestion (non-blocking): thoughts on using an empty set rather than an empty optional if there are no edited programs or questions in the DRAFT version? seems an empty optional and an optional containing an empty set are semantically identical atm

Reply:
Ah I see, that's reasonable